lib/ansible/plugins/__init__.py

```diff
@@ -100,3 +100,29 @@ def has_option(self, option):
     def _check_required(self):
         # FIXME: standardize required check based on config
         pass
+
+    def match_name(self, list_of_names):
+        ''' See if plugin matches any names from a given list of names '''
+
+        found = None
+        mynames = set([self._load_name])  # start with the default
+
+        # all possible intermediates
+        for attrib in ('NAME', 'resolved_name', 'redirected_names'):
```

Comment:
I don't think we should include NAME or get the attributes directly from the plugin, a plugin could lie. I added parsing for _load_name in https://github.com/ansible/ansible/pull/78562/files but that will mean changing _load_name to be the fqcn will also need to update how vars plugins are handled. Or it could be added first and I'd refactor my PR to use it.